### PR TITLE
test: add support for all networks in `deserialize_v2`

### LINKED ISSUE #27140: Add support for all networks in `deserialize_v2` in test_framework
`deserialize_v2` function (`test_framework`) should be able to deseriablize all networks presented in BIP155 (e.g. onion, ipv6, etc), however, it only works for ipv4 and i2p. I noticed that when I was using `message-capture-parser` and realized that most `addrv2` messages weren't able to be deserialized.

### PR DESCRIPTION
**Pull request by brunoerg**
Fixes #27140

### CODE REVIEW

**Comment by fanquake** [comment]
https://cirrus-ci.com/task/6600092300345344
```bash
 test  2023-03-21T23:55:59.273000Z TestFramework (ERROR): Unexpected exception caught during testing 
                                   Traceback (most recent call last):
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 132, in main
                                       self.run_test()
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\p2p_segwit.py", line 285, in run_test
                                       self.test_standardness_v0()
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\p2p_segwit.py", line 112, in func_wrapper
                                       func(self, *args, **kwargs)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\p2p_segwit.py", line 660, in test_standardness_v0
                                       self.generate(self.nodes[0], 1)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 665, in generate
                                       sync_fun() if sync_fun else self.sync_all()
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 729, in sync_all
                                       self.sync_blocks(nodes)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 694, in sync_blocks
                                       best_hash = [x.getbestblockhash() for x in rpc_connections]
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 694, in <listcomp>
                                       best_hash = [x.getbestblockhash() for x in rpc_connections]
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\coverage.py", line 49, in __call__
                                       return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\authproxy.py", line 147, in __call__
                                       response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\authproxy.py", line 110, in _request
                                       self.__conn.request(method, path, postdata, headers)
                                     File "C:\Python39\lib\http\client.py", line 1257, in request
                                       self._send_request(method, url, body, headers, encode_chunked)
                                     File "C:\Python39\lib\http\client.py", line 1303, in _send_request
                                       self.endheaders(body, encode_chunked=encode_chunked)
                                     File "C:\Python39\lib\http\client.py", line 1252, in endheaders
                                       self._send_output(message_body, encode_chunked=encode_chunked)
                                     File "C:\Python39\lib\http\client.py", line 1012, in _send_output
                                       self.send(msg)
                                     File "C:\Python39\lib\http\client.py", line 952, in send
                                       self.connect()
                                     File "C:\Python39\lib\http\client.py", line 925, in connect
                                       self.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
                                   OSError: [WinError 10022] An invalid argument was supplied
 test  2023-03-21T23:55:59.280000Z TestFramework (DEBUG): Closing down network thread 
```

**Comment by brunoerg** [comment]
@fanquake That error seems unrelated to this PR

File: test/functional/test_framework/messages.py [hunk 1]
```diff
@@ -293,17 +305,24 @@ def deserialize_v2(self, f):
...
-        else:
+        elif self.net == self.NET_IPV6:
+            self.ip = socket.inet_ntop(socket.AF_INET6, addr_bytes)
+        elif self.net == self.NET_I2P:
             self.ip = b32encode(addr_bytes)[0:-len(self.I2P_PAD)].decode("ascii").lower() + ".b32.i2p"
-
+        elif self.net == self.NET_CJDNS:
+           self.ip = ":".join(["{:02x}".format(b) for b in addr_bytes])
+        elif self.net in (self.NET_TORV2, self.NET_TORV3):
+            self.ip = b32encode(addr_bytes).decode('ascii') + '.onion'
```
Comment (pinheadmz) [inline test/functional/test_framework/messages.py:325]: Is this correct for torv3? I had to add version and checksum in https://github.com/bitcoin/bitcoin/pull/27452

Comment (brunoerg) [reply test/functional/test_framework/messages.py:329]: You're correct, seems better your approach.

**Comment by brunoerg** [comment]
Force-pushed changing the approach for torv3 addresses, added @pinheadmz as co-author.

**Review by jonatack — COMMENTED** [review]
Concept ACK. This is very similar to a subset of the work in #27452, though a bit less complete here for the `messages.py` changes.

I'm not sure how these changes were coordinated/worked on, but maybe this could be a separate commit in the other pull, which could possibly be split into preparatory and implementation commits.

**Comment by brunoerg** [comment]
Closing this in favor of #27452

**Pull request closed without merging**